overlord/devicestate/handlers_gadget.go

```diff
@@ -138,9 +138,16 @@ func (m *DeviceManager) doUpdateGadgetAssets(t *state.Task, _ *tomb.Tomb) error
 		updatePolicy = gadget.RemodelUpdatePolicy
 	}
 
-	observeTrustedBootAssets := boot.TrustedAssetsUpdateObserverForModel(model)
+	var updateObserver gadget.ContentUpdateObserver
+	observeTrustedBootAssets, err := boot.TrustedAssetsUpdateObserverForModel(model)
+	if err != nil && err != boot.ErrObserverNotApplicable {
+		return err
```

Comment:
this can't happen atm, otoh for consistency better to add some fmt.Errorf wrapping also here

Reply:
If this branch gets green, I'll add the tweak in a followup PR.